test/e2e/run_passwd_test.go

```diff
@@ -90,13 +90,6 @@ USER 1000`, ALPINE)
 		Expect(session.OutputToString()).To(ContainSubstring("/etc/group"))
 	})
 
-	It("podman run numeric user not specified in container modifies group", func() {
-		session := podmanTest.Podman([]string{"run", "--read-only", "-u", "20001", BB, "mount"})
-		session.WaitWithDefaultTimeout()
-		Expect(session).Should(ExitCleanly())
-		Expect(session.OutputToString()).To(ContainSubstring("/etc/group"))
-	})
```

Comment:
Isn't this just a straight up breaking change? We could have people that depend on the group being set or am I missing something?

I agree that this test is bad as it should really check the contents of /etc/group to ensure the id is actually set.

Reply:
it is a "breaking change" but the old behavior doesn't make sense to me.

Why do we create a group with the value of UID if nobody asked for it?

```
➜ podman run --rm --user 1234 alpine grep 1234 /etc/group
1234:x:1234:1234
```

now it seems better to me:

```
➜ bin/podman run --rm --user 1234 alpine grep 1234 /etc/group
➜
```

Reply:
yeah the behavior doesn't seem to make a lot of sense but since there is a test for it I assume it was intentional? https://github.com/containers/podman/commit/f57c39fc7c32000f5f16e3a7813c1b0d6f5af2aa

That said looking at this closer we do not add the user to the group 1234 it keeps using group 0. As such adding the same group to /etc/group does indeed seem to be wrong as it doesn't do anything useful and is just confusing.
So I agree we can call this a bug fix as the group stayed unused nobody should actually depend on it.

cc @mheon

Reply:
2020 is a long time ago now, but I assume this was in response to #7389 - which makes sense (if we are adding a group, it really ought to be in `/etc/group`) but only if the user is actually in the group that was being added. I assume that `--user 1234` automatically created a group with ID 1234 at one point and added the user to it, and that was eventually changed to not do so, but we never altered how /etc/passwd worked?
I think the change is OK to go through because of this, looks like a remnant of a long ago refactoring.